diff --git a/ltmain.sh b/ltmain.sh
index 6939dcc4a662..9ccc3997eb48 100755
--- a/ltmain.sh
+++ b/ltmain.sh
@@ -5785,7 +5785,7 @@ func_mode_link ()
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir="-L$libdir"
+	      add_dir="-L$lt_sysroot$libdir"
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
