diff --git a/tests/Gtest-nomalloc.c b/tests/Gtest-nomalloc.c
index 5b97fc70918c..42bc80febc4e 100644
--- a/tests/Gtest-nomalloc.c
+++ b/tests/Gtest-nomalloc.c
@@ -37,10 +37,10 @@ int in_unwind;
 void *
 malloc(size_t s)
 {
-  static void * (*func)();
+  static void * (*func)(size_t);
 
   if(!func)
-    func = (void *(*)()) dlsym(RTLD_NEXT, "malloc");
+    func = (void *(*)(size_t)) dlsym(RTLD_NEXT, "malloc");
 
   if (in_unwind) {
     num_errors++;
