diff --git a/support/export/Makefile.am b/support/export/Makefile.am
index 13f7a49cefb6..b39ecc4094cd 100644
--- a/support/export/Makefile.am
+++ b/support/export/Makefile.am
@@ -1,5 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
 
 GENFILES_CLNT	= mount_clnt.c
 GENFILES_XDR	= mount_xdr.c
@@ -24,7 +25,7 @@ dist-hook:
 if CONFIG_RPCGEN
 RPCGEN		= $(top_builddir)/tools/rpcgen/rpcgen
 $(RPCGEN):
-	make -C $(top_srcdir)/tools/rpcgen all
+	make -C $(top_builddir)/tools/rpcgen all
 else
 RPCGEN = @RPCGEN_PATH@
 endif
diff --git a/support/misc/Makefile.am b/support/misc/Makefile.am
index f9993e3ac897..e1f94fe5d021 100644
--- a/support/misc/Makefile.am
+++ b/support/misc/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include
+
 noinst_LIBRARIES = libmisc.a
 libmisc_a_SOURCES = tcpwrapper.c from_local.c mountpoint.c file.c \
 		    nfsd_path.c workqueue.c xstat.c
diff --git a/support/nfs/Makefile.am b/support/nfs/Makefile.am
index 67e3a8e134c8..63c983fd4559 100644
--- a/support/nfs/Makefile.am
+++ b/support/nfs/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
+
 noinst_LIBRARIES = 
 noinst_LTLIBRARIES = libnfs.la libnfsconf.la
 
diff --git a/support/nfsidmap/Makefile.am b/support/nfsidmap/Makefile.am
index 9c21fa34febb..47833b41bb73 100644
--- a/support/nfsidmap/Makefile.am
+++ b/support/nfsidmap/Makefile.am
@@ -1,3 +1,5 @@
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
+
 if PATH_PLUGINS
 pkgplugindir=$(PATH_PLUGINS)
 else
diff --git a/support/nsm/Makefile.am b/support/nsm/Makefile.am
index 8f5874ef26b8..1e3eff599358 100644
--- a/support/nsm/Makefile.am
+++ b/support/nsm/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
+
 GENFILES_CLNT	= sm_inter_clnt.c
 GENFILES_SVC	= sm_inter_svc.c
 GENFILES_XDR	= sm_inter_xdr.c
diff --git a/tools/locktest/Makefile.am b/tools/locktest/Makefile.am
index 3156815d4ac3..95580555d226 100644
--- a/tools/locktest/Makefile.am
+++ b/tools/locktest/Makefile.am
@@ -1,6 +1,9 @@
 ## Process this file with automake to produce Makefile.in
 
 CC=$(CC_FOR_BUILD)
+CFLAGS=$(CFLAGS_FOR_BUILD)
+CPPFLAGS=$(CPPFLAGS_FOR_BUILD)
+LDFLAGS=$(LDFLAGS_FOR_BUILD)
 LIBTOOL = @LIBTOOL@ --tag=CC
 
 noinst_PROGRAMS = testlk
diff --git a/tools/mountstats/Makefile.am b/tools/mountstats/Makefile.am
index c2e9f9995ada..86ae4398d73f 100644
--- a/tools/mountstats/Makefile.am
+++ b/tools/mountstats/Makefile.am
@@ -8,6 +8,6 @@ EXTRA_DIST	= $(man8_MANS) $(PYTHON_FILES)
 all-local: $(PYTHON_FILES)
 
 install-data-hook:
-	$(INSTALL) -m 755 mountstats.py $(DESTDIR)$(sbindir)/mountstats
+	$(INSTALL) -m 755 $(srcdir)/mountstats.py $(DESTDIR)$(sbindir)/mountstats
 
 MAINTAINERCLEANFILES=Makefile.in
diff --git a/tools/nfs-iostat/Makefile.am b/tools/nfs-iostat/Makefile.am
index 3ae0f29701ca..c939d7e14d76 100644
--- a/tools/nfs-iostat/Makefile.am
+++ b/tools/nfs-iostat/Makefile.am
@@ -8,6 +8,6 @@ EXTRA_DIST	= $(man8_MANS) $(PYTHON_FILES)
 all-local: $(PYTHON_FILES)
 
 install-data-hook:
-	$(INSTALL) -m 755 nfs-iostat.py $(DESTDIR)$(sbindir)/nfsiostat
+	$(INSTALL) -m 755 $(srcdir)/nfs-iostat.py $(DESTDIR)$(sbindir)/nfsiostat
 
 MAINTAINERCLEANFILES=Makefile.in
diff --git a/tools/nfsconf/Makefile.am b/tools/nfsconf/Makefile.am
index b3c1495f83cd..aa7428d492e5 100644
--- a/tools/nfsconf/Makefile.am
+++ b/tools/nfsconf/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
+
 man8_MANS	= nfsconf.man
 EXTRA_DIST	= $(man8_MANS)
 
diff --git a/tools/rpcdebug/Makefile.am b/tools/rpcdebug/Makefile.am
index b0a3e1f9fe1a..9f80c45c6095 100644
--- a/tools/rpcdebug/Makefile.am
+++ b/tools/rpcdebug/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include
+
 man8_MANS = rpcdebug.man
 EXTRA_DIST = $(man8_MANS)
 
diff --git a/tools/rpcgen/Makefile.am b/tools/rpcgen/Makefile.am
index 8a9ec89cea76..b2d43fc76ff8 100644
--- a/tools/rpcgen/Makefile.am
+++ b/tools/rpcgen/Makefile.am
@@ -1,6 +1,9 @@
 ## Process this file with automake to produce Makefile.in
 
 CC=$(CC_FOR_BUILD)
+CFLAGS=$(CFLAGS_FOR_BUILD)
+CPPFLAGS=$(CPPFLAGS_FOR_BUILD)
+LDFLAGS=$(LDFLAGS_FOR_BUILD)
 LIBTOOL = @LIBTOOL@ --tag=CC
 
 noinst_PROGRAMS = rpcgen
diff --git a/utils/blkmapd/Makefile.am b/utils/blkmapd/Makefile.am
index 56c8a4bb4ee2..8d6f12fff399 100644
--- a/utils/blkmapd/Makefile.am
+++ b/utils/blkmapd/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
+
 man8_MANS	= blkmapd.man
 EXTRA_DIST = $(man8_MANS)
 
diff --git a/utils/exportfs/Makefile.am b/utils/exportfs/Makefile.am
index 96524c729359..e2d461d735dd 100644
--- a/utils/exportfs/Makefile.am
+++ b/utils/exportfs/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
+
 man5_MANS	= exports.man
 man7_MANS	= nfsd.man
 man8_MANS	= exportfs.man
diff --git a/utils/idmapd/Makefile.am b/utils/idmapd/Makefile.am
index e09e8c53821d..3727da5a8bf4 100644
--- a/utils/idmapd/Makefile.am
+++ b/utils/idmapd/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
+
 man8_MANS	= idmapd.man
 
 AM_CPPFLAGS += -I ../../support/nfsidmap
diff --git a/utils/mount/Makefile.am b/utils/mount/Makefile.am
index ad0be93b1def..79402b7f1890 100644
--- a/utils/mount/Makefile.am
+++ b/utils/mount/Makefile.am
@@ -1,8 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-# These binaries go in /sbin (not /usr/sbin), and that cannot be
-# overridden at config time.
-sbindir = /sbin
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
 
 man8_MANS	= mount.nfs.man umount.nfs.man
 man5_MANS	= nfs.man
diff --git a/utils/mountd/Makefile.am b/utils/mountd/Makefile.am
index 18610f18238c..c452a6502bdf 100644
--- a/utils/mountd/Makefile.am
+++ b/utils/mountd/Makefile.am
@@ -5,6 +5,8 @@ if CONFIG_JUNCTION
 OPTLIBS		+= ../../support/junction/libjunction.la $(LIBXML2)
 endif
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
+
 man8_MANS	= mountd.man
 EXTRA_DIST	= $(man8_MANS)
 
diff --git a/utils/nfsd/Makefile.am b/utils/nfsd/Makefile.am
index 8acc9a04a8c0..4d6fb184a7af 100644
--- a/utils/nfsd/Makefile.am
+++ b/utils/nfsd/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
+
 man8_MANS	= nfsd.man
 EXTRA_DIST	= $(man8_MANS)
 
diff --git a/utils/nfsdcld/Makefile.am b/utils/nfsdcld/Makefile.am
index 2c4e5a11c417..86ebe82cdd32 100644
--- a/utils/nfsdcld/Makefile.am
+++ b/utils/nfsdcld/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
+
 # These binaries go in /sbin (not /usr/sbin), and that cannot be
 # overridden at config time. The kernel "knows" the /sbin name.
 sbindir = /sbin
diff --git a/utils/nfsdcltrack/Makefile.am b/utils/nfsdcltrack/Makefile.am
index 2f7fe3de6922..494993280077 100644
--- a/utils/nfsdcltrack/Makefile.am
+++ b/utils/nfsdcltrack/Makefile.am
@@ -1,8 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-# These binaries go in /sbin (not /usr/sbin), and that cannot be
-# overridden at config time. The kernel "knows" the /sbin name.
-sbindir = /sbin
+AM_CPPFLAGS = -I $(top_srcdir)/support/include
 
 man8_MANS	= nfsdcltrack.man
 EXTRA_DIST	= $(man8_MANS)
diff --git a/utils/nfsidmap/Makefile.am b/utils/nfsidmap/Makefile.am
index e5d7d04c1799..6ccd07327937 100644
--- a/utils/nfsidmap/Makefile.am
+++ b/utils/nfsidmap/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
+
 man8_MANS = nfsidmap.man
 sbin_PROGRAMS	= nfsidmap
 
diff --git a/utils/showmount/Makefile.am b/utils/showmount/Makefile.am
index d0a16b2b048a..28b582025257 100644
--- a/utils/showmount/Makefile.am
+++ b/utils/showmount/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
+
 man8_MANS	= showmount.man
 EXTRA_DIST	= $(man8_MANS)
 
diff --git a/utils/statd/Makefile.am b/utils/statd/Makefile.am
index 6facc15d856c..3153b5014faa 100644
--- a/utils/statd/Makefile.am
+++ b/utils/statd/Makefile.am
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+AM_CPPFLAGS = -I $(top_srcdir)/support/include $(TIRPC_CFLAGS)
+
 man8_MANS = statd.man sm-notify.man
 
 RPCPREFIX	= rpc.
