diff --git a/utils/blkmapd/device-discovery.c b/utils/blkmapd/device-discovery.c
index 8eb3fd0a56bd..3a1342e2354a 100644
--- a/utils/blkmapd/device-discovery.c
+++ b/utils/blkmapd/device-discovery.c
@@ -32,6 +32,7 @@
 #include <sys/mount.h>
 #include <sys/select.h>
 #include <sys/inotify.h>
+#include <sys/sysmacros.h>
 #include <linux/kdev_t.h>
 #include <scsi/scsi.h>
 #include <scsi/scsi_ioctl.h>
