commit beba9639050bcc9f2cac7e0a797f161a4434cf54
Author: Thierry Reding <treding@nvidia.com>
Date:   Tue Nov 29 14:17:46 2022 +0100

    no-statedir

diff --git a/Makefile.am b/Makefile.am
index 00220842e2d0..f36ba72f2ec0 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -20,18 +20,3 @@ EXTRA_DIST = \
 	aclocal/ipv6.m4
 
 ACLOCAL_AMFLAGS = -I aclocal
-
-install-data-hook:
-	if [ ! -d $(DESTDIR)$(statedir) ]; then mkdir -p $(DESTDIR)$(statedir); fi
-	touch $(DESTDIR)$(statedir)/etab; chmod 644 $(DESTDIR)$(statedir)/etab
-	touch $(DESTDIR)$(statedir)/rmtab; chmod 644 $(DESTDIR)$(statedir)/rmtab
-	mkdir -p $(DESTDIR)$(statdpath)/sm $(DESTDIR)$(statdpath)/sm.bak
-	touch $(DESTDIR)$(statdpath)/state
-	chmod go-rwx $(DESTDIR)$(statdpath)/sm $(DESTDIR)$(statdpath)/sm.bak $(DESTDIR)$(statdpath)/state
-	-chown $(statduser) $(DESTDIR)$(statdpath)/sm $(DESTDIR)$(statdpath)/sm.bak $(DESTDIR)$(statdpath)/state
-
-uninstall-hook:
-	rm -f $(DESTDIR)$(statedir)/xtab
-	rm $(DESTDIR)$(statedir)/etab
-	rm $(DESTDIR)$(statedir)/rmtab
-	rm $(DESTDIR)$(statdpath)/state
