commit f983fd777c29d672be78fadfbfc43eca1fcc4c44
Author: Thierry Reding <treding@nvidia.com>
Date:   Tue Nov 29 14:28:43 2022 +0100

    std-sbindir

diff --git a/tools/rpcctl/Makefile.am b/tools/rpcctl/Makefile.am
index bb70d6fe4dc0..4040ba762a1d 100644
--- a/tools/rpcctl/Makefile.am
+++ b/tools/rpcctl/Makefile.am
@@ -1,3 +1,5 @@
+@CONFIG_SBIN_OVERRIDE_TRUE@sbindir = /sbin
+
 ## Process this file with automake to produce Makefile.in
 PYTHON_FILES =  rpcctl.py
 
