diff --git a/tools/nfsrahead/Makefile.am b/tools/nfsrahead/Makefile.am
index 578e3d640649..673740cb378f 100644
--- a/tools/nfsrahead/Makefile.am
+++ b/tools/nfsrahead/Makefile.am
@@ -8,7 +8,7 @@ nfsrahead_LDADD = ../../support/nfs/libnfsconf.la
 man5_MANS = nfsrahead.man
 EXTRA_DIST = $(man5_MANS)
 
-udev_rulesdir = /usr/lib/udev/rules.d/
+udev_rulesdir = ${libdir}/udev/rules.d/
 udev_rules_DATA = 99-nfs.rules
 
 99-nfs.rules: 99-nfs.rules.in $(builddefs)
