diff --git a/CMakeLists.txt b/CMakeLists.txt
index 69a249b5ad29..83778585d4fa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -143,12 +143,6 @@ if(UHDR_BUILD_BENCHMARK AND EMSCRIPTEN)
   message(FATAL_ERROR "Building benchmarks not supported for wasm targets")
 endif()
 
-# side effects
-if(CMAKE_CROSSCOMPILING AND UHDR_ENABLE_INSTALL)
-  set(UHDR_ENABLE_INSTALL FALSE) # disable install and uninstall targets during cross compilation.
-  message(STATUS "Install and uninstall targets - Disabled")
-endif()
-
 if(UHDR_BUILD_FUZZERS AND NOT UHDR_BUILD_DEPS)
   set(UHDR_BUILD_DEPS TRUE) # For fuzz testing its best to build all dependencies from source.
                             # This is to instrument dependency libs as well.
