diff --git a/configure.py b/configure.py
index 97b595f9e9f3..cb2f962f9ff1 100755
--- a/configure.py
+++ b/configure.py
@@ -25,6 +25,8 @@ p.add_option('--with-llvm-config', metavar='PATH',
              help='use given llvm-config script')
 p.add_option('--with-llvm-bindir', metavar='PATH',
              help='use given directory to find LLVM binaries')
+p.add_option('--with-llvm-cxxflags', metavar='PATH',
+             help='use the given C++ flags at compilation time')
 p.add_option('--with-cxx-compiler', metavar='PATH',
              help='use given C++ compiler')
 p.add_option('--prefix', metavar='PATH',
@@ -42,6 +44,7 @@ p.add_option('--enable-runtime-subnormal', action="store_true", default=False,
 (options, args) = p.parse_args()
 
 llvm_config_exe = options.with_llvm_config or "llvm-config"
+llvm_cxxflags = options.with_llvm_cxxflags
 llvm_bindir = options.with_llvm_bindir
 
 prefix = options.prefix
@@ -85,8 +88,11 @@ if not llvm_bindir:
 llvm_core_libs = llvm_config(['--libs', 'core', 'bitreader', 'bitwriter']) + ' ' + \
                  llvm_system_libs + ' ' + \
                  llvm_config(['--ldflags'])
-llvm_cxxflags = llvm_config(['--cxxflags']) + ' -fno-exceptions -fno-rtti ' + \
-                '-DHAVE_LLVM=0x{:0=4}'.format(llvm_int_version)
+if not llvm_cxxflags:
+    llvm_cxxflags = llvm_config(['--cxxflags'])
+
+llvm_cxxflags += ' -fno-exceptions -fno-rtti ' + \
+                 '-DHAVE_LLVM=0x{:0=4}'.format(llvm_int_version)
 llvm_libdir = llvm_config(['--libdir'])
 
 llvm_clang = os.path.join(llvm_bindir, 'clang')
