diff --git a/meson.build b/meson.build
index 3012935f8132..c17ea38ad679 100644
--- a/meson.build
+++ b/meson.build
@@ -186,7 +186,7 @@ at_spi_conf.set('ALIGNOF_DBIND_STRUCT', cc.alignment('dbind_struct', prefix: 'ty
 
 # introspection support
 gir = find_program('g-ir-scanner', required : get_option('introspection'))
-have_gir = gir.found() and (not meson.is_cross_build() or get_option('introspection').enabled())
+have_gir = gir.found() and get_option('introspection').allowed()
 
 if not get_option('atk_only')
   pymod = import('python')
