diff --git a/configure b/configure
index 158e76e187b2..8f7233afd87c 100755
--- a/configure
+++ b/configure
@@ -20,10 +20,14 @@ check_toolchain()
     : ${AR=ar}
     : ${CC=gcc}
     : ${YACC=bison}
+    : ${CFLAGS=}
+    : ${LDFLAGS=}
     echo "PKG_CONFIG:=${PKG_CONFIG}" >>$CONFIG
     echo "AR:=${AR}" >>$CONFIG
     echo "CC:=${CC}" >>$CONFIG
     echo "YACC:=${YACC}" >>$CONFIG
+    echo "EXTRA_CFLAGS:=${CFLAGS}" >>$CONFIG
+    echo "LDFLAGS:=${LDFLAGS}" >>$CONFIG
 }
 
 check_xtables()
@@ -55,7 +59,7 @@ int main(int argc, char **argv)
 }
 EOF
 
-    if $CC -I$INCLUDE $IPTC -o $TMPDIR/ipttest $TMPDIR/ipttest.c $IPTL \
+    if $CC $CFLAGS -I$INCLUDE $IPTC -o $TMPDIR/ipttest $TMPDIR/ipttest.c $IPTL \
 	$(${PKG_CONFIG} xtables --cflags --libs) -ldl >/dev/null 2>&1; then
 	echo "TC_CONFIG_XT:=y" >>$CONFIG
 	echo "using xtables"
@@ -90,7 +94,7 @@ int main(int argc, char **argv) {
 
 EOF
 
-    if $CC -I$INCLUDE $IPTC -o $TMPDIR/ipttest $TMPDIR/ipttest.c $IPTL -ldl >/dev/null 2>&1; then
+    if $CC $CFLAGS -I$INCLUDE $IPTC -o $TMPDIR/ipttest $TMPDIR/ipttest.c $IPTL -ldl >/dev/null 2>&1; then
 	echo "TC_CONFIG_XT_OLD:=y" >>$CONFIG
 	echo "using old xtables (no need for xt-internal.h)"
     fi
@@ -124,7 +128,7 @@ int main(int argc, char **argv) {
 }
 
 EOF
-	if $CC -I$INCLUDE $IPTC -o $TMPDIR/ipttest $TMPDIR/ipttest.c $IPTL -ldl >/dev/null 2>&1; then
+	if $CC $CFLAGS -I$INCLUDE $IPTC -o $TMPDIR/ipttest $TMPDIR/ipttest.c $IPTL -ldl >/dev/null 2>&1; then
 	    echo "using old xtables with xt-internal.h"
 	    echo "TC_CONFIG_XT_OLD_H:=y" >>$CONFIG
 	fi
@@ -180,7 +184,7 @@ int main(int argc, char **argv)
 	return 0;
 }
 EOF
-    if $CC -I$INCLUDE -o $TMPDIR/setnstest $TMPDIR/setnstest.c >/dev/null 2>&1; then
+    if $CC $CFLAGS -I$INCLUDE -o $TMPDIR/setnstest $TMPDIR/setnstest.c >/dev/null 2>&1; then
 	echo "IP_CONFIG_SETNS:=y" >>$CONFIG
 	echo "yes"
 	echo "CFLAGS += -DHAVE_SETNS" >>$CONFIG
@@ -235,7 +239,7 @@ int main(void)
 #endif
 EOF
 
-    if $CC -I$INCLUDE -o $TMPDIR/ipsettest $TMPDIR/ipsettest.c >/dev/null 2>&1; then
+    if $CC $CFLAGS -I$INCLUDE -o $TMPDIR/ipsettest $TMPDIR/ipsettest.c >/dev/null 2>&1; then
 	echo "TC_CONFIG_IPSET:=y" >>$CONFIG
 	echo "yes"
     else
@@ -413,7 +417,7 @@ int main(int argc, char **argv) {
 	return 0;
 }
 EOF
-    if $CC -I$INCLUDE -o $TMPDIR/dbtest $TMPDIR/dbtest.c -ldb >/dev/null 2>&1; then
+    if $CC $CFLAGS -I$INCLUDE -o $TMPDIR/dbtest $TMPDIR/dbtest.c -ldb >/dev/null 2>&1; then
 	echo "HAVE_BERKELEY_DB:=y" >>$CONFIG
 	echo "yes"
     else
@@ -432,7 +436,7 @@ int main(int argc, char **argv) {
 	return 0;
 }
 EOF
-    if $CC -I$INCLUDE -o $TMPDIR/strtest $TMPDIR/strtest.c >/dev/null 2>&1; then
+    if $CC $CFLAGS -I$INCLUDE -o $TMPDIR/strtest $TMPDIR/strtest.c >/dev/null 2>&1; then
 	echo "no"
     else
 	if ${PKG_CONFIG} libbsd --exists; then
