diff --git a/Make.Rules b/Make.Rules
index cc6f95b6e5b9..d19fa811fe70 100644
--- a/Make.Rules
+++ b/Make.Rules
@@ -57,7 +57,7 @@ IPATH += -fPIC -I$(KERNEL_HEADERS) -I$(topdir)/libcap/include
 CC := $(CROSS_COMPILE)gcc
 DEFINES := -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 COPTS ?= -O2
-CFLAGS ?= $(COPTS) $(DEFINES)
+CFLAGS ?= $(COPTS) $(DEFINES) $(EXTRA_CFLAGS)
 BUILD_CC ?= $(CC)
 BUILD_COPTS ?= -O2
 BUILD_CFLAGS ?= $(BUILD_COPTS) $(DEFINES) $(IPATH)
@@ -69,7 +69,7 @@ WARNINGS=-Wall -Wwrite-strings \
         -Wstrict-prototypes -Wmissing-prototypes \
         -Wnested-externs -Winline -Wshadow
 LD=$(CC) -Wl,-x -shared
-LDFLAGS ?= #-g
+LDFLAGS ?= $(EXTRA_LDFLAGS) #-g
 LIBCAPLIB := -L$(topdir)/libcap -lcap
 PSXLINKFLAGS :=  -lpthread -Wl,-wrap,pthread_create
 LIBPSXLIB := -L$(topdir)/libcap -lpsx $(PSXLINKFLAGS)
