diff --git a/Makefile b/Makefile
index c4abc71..366b7f8 100644
--- a/Makefile
+++ b/Makefile
@@ -60,7 +60,8 @@ OPENSSLDIR=/usr/local/ssl
 # PKCS1_CHECK - pkcs1 tests.
 
 CC= cc
-CFLAG= -O
+CFLAG=
+CFLAGS=$(EXTRA_CPPFLAGS) $(CFLAG) $(EXTRA_CFLAGS)
 DEPFLAG= -DOPENSSL_NO_EC_NISTP_64_GCC_128 -DOPENSSL_NO_GMP -DOPENSSL_NO_JPAKE -DOPENSSL_NO_LIBUNBOUND -DOPENSSL_NO_MD2 -DOPENSSL_NO_RC5 -DOPENSSL_NO_RFC3779 -DOPENSSL_NO_SCTP -DOPENSSL_NO_SSL_TRACE -DOPENSSL_NO_SSL2 -DOPENSSL_NO_STORE -DOPENSSL_NO_UNIT_TEST -DOPENSSL_NO_WEAK_SSL_CIPHERS
 PEX_LIBS= 
 EX_LIBS= 
@@ -298,7 +299,7 @@ build_testapps:
 	@dir=crypto; target=testapps; $(BUILD_ONE_CMD)
 
 fips_premain_dso$(EXE_EXT): libcrypto.a
-	[ -z "$(FIPSCANLIB)" ] || $(CC) $(CFLAG) -Iinclude \
+	[ -z "$(FIPSCANLIB)" ] || $(CC) $(CFLAGS) -Iinclude \
 		-DFINGERPRINT_PREMAIN_DSO_LOAD -o $@  \
 		$(FIPSLIBDIR)fips_premain.c $(FIPSLIBDIR)fipscanister.o \
 		libcrypto.a $(EX_LIBS)
diff --git a/Makefile.shared b/Makefile.shared
index 5b960d9cdaec..b78ea7ecb7da 100644
--- a/Makefile.shared
+++ b/Makefile.shared
@@ -7,7 +7,7 @@
 
 # CC contains the current compiler.  This one MUST be defined
 CC=cc
-CFLAGS=$(CFLAG)
+CFLAGS=$(EXTRA_CPPFLAGS) $(CFLAG) $(EXTRA_CFLAGS)
 # LDFLAGS contains flags to be used when temporary object files (when building
 # shared libraries) are created, or when an application is linked.
 # SHARED_LDFLAGS contains flags to be used when the shared library is created.
diff --git a/apps/Makefile b/apps/Makefile
index 8c3297ea6724..a54bdb570a82 100644
--- a/apps/Makefile
+++ b/apps/Makefile
@@ -20,7 +20,7 @@ EXE_EXT=
 
 SHLIB_TARGET=
 
-CFLAGS= -DMONOLITH $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) -DMONOLITH $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile makeapps.com install.com
 
diff --git a/crypto/Makefile b/crypto/Makefile
index 7869996a9c07..3d89f7d7e3fb 100644
--- a/crypto/Makefile
+++ b/crypto/Makefile
@@ -23,7 +23,7 @@ RECURSIVE_MAKE=	[ -n "$(SDIRS)" ] && for i in $(SDIRS) ; do \
 PEX_LIBS=
 EX_LIBS=
  
-CFLAGS= $(INCLUDE) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDE) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDE) $(ASFLAG)
 AFLAGS=$(ASFLAGS)
 CPUID_OBJ=mem_clr.o
diff --git a/crypto/aes/Makefile b/crypto/aes/Makefile
index e825c140194f..bbfa1f1b0d58 100644
--- a/crypto/aes/Makefile
+++ b/crypto/aes/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 AES_ENC=aes_core.o aes_cbc.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/asn1/Makefile b/crypto/asn1/Makefile
index 330fe81b740c..0dba731bc7fa 100644
--- a/crypto/asn1/Makefile
+++ b/crypto/asn1/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=
diff --git a/crypto/bf/Makefile b/crypto/bf/Makefile
index 6dd2015537fe..fb3f8fd5ab07 100644
--- a/crypto/bf/Makefile
+++ b/crypto/bf/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 BF_ENC=		bf_enc.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/bio/Makefile b/crypto/bio/Makefile
index ef526f6beb2c..435d5cda2975 100644
--- a/crypto/bio/Makefile
+++ b/crypto/bio/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/bn/Makefile b/crypto/bn/Makefile
index c4c640951759..ba4c9f8abcfe 100644
--- a/crypto/bn/Makefile
+++ b/crypto/bn/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 BN_ASM=		bn_asm.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/buffer/Makefile b/crypto/buffer/Makefile
index 352efb841ffb..0c36748cfb21 100644
--- a/crypto/buffer/Makefile
+++ b/crypto/buffer/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/camellia/Makefile b/crypto/camellia/Makefile
index ab1225e7d902..5acec799df32 100644
--- a/crypto/camellia/Makefile
+++ b/crypto/camellia/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 CMLL_ENC= camellia.o cmll_misc.o cmll_cbc.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/cast/Makefile b/crypto/cast/Makefile
index 4c4b5e9baa55..60e47ea570e2 100644
--- a/crypto/cast/Makefile
+++ b/crypto/cast/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 CAST_ENC=c_enc.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/cmac/Makefile b/crypto/cmac/Makefile
index 6a2840867643..799a34a63099 100644
--- a/crypto/cmac/Makefile
+++ b/crypto/cmac/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(INCLUDES) $(EXTRA_CFLAGS) $(CFLAG)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/cms/Makefile b/crypto/cms/Makefile
index 6f3a83202638..34d367a1a6b4 100644
--- a/crypto/cms/Makefile
+++ b/crypto/cms/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/comp/Makefile b/crypto/comp/Makefile
index a1e9464a11f7..a8adf21848d6 100644
--- a/crypto/comp/Makefile
+++ b/crypto/comp/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/conf/Makefile b/crypto/conf/Makefile
index d5f5c582414b..4535446aaf33 100644
--- a/crypto/conf/Makefile
+++ b/crypto/conf/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/des/Makefile b/crypto/des/Makefile
index 8b5166ca9ff5..3783f7eac6c9 100644
--- a/crypto/des/Makefile
+++ b/crypto/des/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 RANLIB=		ranlib
 DES_ENC=	des_enc.o fcrypt_b.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/dh/Makefile b/crypto/dh/Makefile
index 46fa5ac57b47..ade1e585fa4d 100644
--- a/crypto/dh/Makefile
+++ b/crypto/dh/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST= dhtest.c
diff --git a/crypto/dsa/Makefile b/crypto/dsa/Makefile
index 81092013773f..7614aee25341 100644
--- a/crypto/dsa/Makefile
+++ b/crypto/dsa/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=dsatest.c
diff --git a/crypto/dso/Makefile b/crypto/dso/Makefile
index 36b8ead041c3..acc77cf8c0c6 100644
--- a/crypto/dso/Makefile
+++ b/crypto/dso/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/ec/Makefile b/crypto/ec/Makefile
index 89491454a441..474befa41e18 100644
--- a/crypto/ec/Makefile
+++ b/crypto/ec/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/ecdh/Makefile b/crypto/ecdh/Makefile
index 1b31ba1f0b3f..732f1d668c83 100644
--- a/crypto/ecdh/Makefile
+++ b/crypto/ecdh/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g -Wall
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=ecdhtest.c
diff --git a/crypto/ecdsa/Makefile b/crypto/ecdsa/Makefile
index 4ce00e8f9308..6c97f2016b93 100644
--- a/crypto/ecdsa/Makefile
+++ b/crypto/ecdsa/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g -Wall
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=ecdsatest.c
diff --git a/crypto/engine/Makefile b/crypto/engine/Makefile
index 426388e9b174..bf1e46089b18 100644
--- a/crypto/engine/Makefile
+++ b/crypto/engine/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST= enginetest.c
diff --git a/crypto/err/Makefile b/crypto/err/Makefile
index b6f3ef1778d1..edfeb984a567 100644
--- a/crypto/err/Makefile
+++ b/crypto/err/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/evp/Makefile b/crypto/evp/Makefile
index fa138d0b1014..f3a052dd55fb 100644
--- a/crypto/evp/Makefile
+++ b/crypto/evp/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=evp_test.c evp_extra_test.c
diff --git a/crypto/hmac/Makefile b/crypto/hmac/Makefile
index 52e39e586075..c8bd709cfc52 100644
--- a/crypto/hmac/Makefile
+++ b/crypto/hmac/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=hmactest.c
diff --git a/crypto/idea/Makefile b/crypto/idea/Makefile
index 3dc23e48dd42..d3f10f46a439 100644
--- a/crypto/idea/Makefile
+++ b/crypto/idea/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=ideatest.c
diff --git a/crypto/jpake/Makefile b/crypto/jpake/Makefile
index 5193fd983533..7722f2695328 100644
--- a/crypto/jpake/Makefile
+++ b/crypto/jpake/Makefile
@@ -1,7 +1,7 @@
 DIR=jpake
 TOP=../..
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 LIB=$(TOP)/libcrypto.a
 LIBOBJ=jpake.o jpake_err.o
diff --git a/crypto/krb5/Makefile b/crypto/krb5/Makefile
index 8b9a01a296c1..2fbdb8de65d5 100644
--- a/crypto/krb5/Makefile
+++ b/crypto/krb5/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=
diff --git a/crypto/lhash/Makefile b/crypto/lhash/Makefile
index c7f4365f0ab9..06ab070651f6 100644
--- a/crypto/lhash/Makefile
+++ b/crypto/lhash/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/md2/Makefile b/crypto/md2/Makefile
index b63011085f89..868f109ce542 100644
--- a/crypto/md2/Makefile
+++ b/crypto/md2/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=md2test.c
diff --git a/crypto/md4/Makefile b/crypto/md4/Makefile
index 3ee436176ba6..fa94df23c1bc 100644
--- a/crypto/md4/Makefile
+++ b/crypto/md4/Makefile
@@ -11,7 +11,7 @@ CFLAG=-g
 MAKEFILE=       Makefile
 AR=             ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=md4test.c
diff --git a/crypto/md5/Makefile b/crypto/md5/Makefile
index f5240da74cd7..b8bfc522a117 100644
--- a/crypto/md5/Makefile
+++ b/crypto/md5/Makefile
@@ -13,7 +13,7 @@ AR=             ar r
 
 MD5_ASM_OBJ=
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/mdc2/Makefile b/crypto/mdc2/Makefile
index c2d0c5b7cd58..73400827600e 100644
--- a/crypto/mdc2/Makefile
+++ b/crypto/mdc2/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST= mdc2test.c
diff --git a/crypto/modes/Makefile b/crypto/modes/Makefile
index a7863d98be2f..cebc6bc9aeb1 100644
--- a/crypto/modes/Makefile
+++ b/crypto/modes/Makefile
@@ -12,7 +12,7 @@ AR=		ar r
 
 MODES_ASM_OBJ=
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/objects/Makefile b/crypto/objects/Makefile
index f93d2f9d2647..b46df5fc95bf 100644
--- a/crypto/objects/Makefile
+++ b/crypto/objects/Makefile
@@ -11,7 +11,7 @@ MAKEFILE=	Makefile
 AR=		ar r
 PERL=		perl
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=
diff --git a/crypto/ocsp/Makefile b/crypto/ocsp/Makefile
index 96a1b156b9fb..14de97fa906f 100644
--- a/crypto/ocsp/Makefile
+++ b/crypto/ocsp/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=
diff --git a/crypto/pem/Makefile b/crypto/pem/Makefile
index 65de60e2a12e..48a39ef33e67 100644
--- a/crypto/pem/Makefile
+++ b/crypto/pem/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/pkcs12/Makefile b/crypto/pkcs12/Makefile
index be5f8c5d21df..763d111ac073 100644
--- a/crypto/pkcs12/Makefile
+++ b/crypto/pkcs12/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/pkcs7/Makefile b/crypto/pkcs7/Makefile
index decf5e0203e8..49a62841f266 100644
--- a/crypto/pkcs7/Makefile
+++ b/crypto/pkcs7/Makefile
@@ -12,8 +12,8 @@ AR=		ar r
 
 PEX_LIBS=
 EX_LIBS=
- 
-CFLAGS= $(INCLUDES) $(CFLAG)
+
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=
diff --git a/crypto/pqueue/Makefile b/crypto/pqueue/Makefile
index a59b5a9395b2..86f094894d33 100644
--- a/crypto/pqueue/Makefile
+++ b/crypto/pqueue/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/rand/Makefile b/crypto/rand/Makefile
index df44369a0823..da73a3ca45e7 100644
--- a/crypto/rand/Makefile
+++ b/crypto/rand/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST= randtest.c
diff --git a/crypto/rc2/Makefile b/crypto/rc2/Makefile
index b3727a4a6d47..6ef001b0ad66 100644
--- a/crypto/rc2/Makefile
+++ b/crypto/rc2/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=rc2test.c
diff --git a/crypto/rc4/Makefile b/crypto/rc4/Makefile
index 7434ff737e66..c23b365c4691 100644
--- a/crypto/rc4/Makefile
+++ b/crypto/rc4/Makefile
@@ -12,7 +12,7 @@ AR=		ar r
 
 RC4_ENC=rc4_enc.o rc4_skey.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/rc5/Makefile b/crypto/rc5/Makefile
index 6ca0037c681b..fa9acefffdbe 100644
--- a/crypto/rc5/Makefile
+++ b/crypto/rc5/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 RC5_ENC=		rc5_enc.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/ripemd/Makefile b/crypto/ripemd/Makefile
index 1c3f094bb088..44b59c4e2f25 100644
--- a/crypto/ripemd/Makefile
+++ b/crypto/ripemd/Makefile
@@ -13,7 +13,7 @@ AR=             ar r
 
 RIP_ASM_OBJ=
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/rsa/Makefile b/crypto/rsa/Makefile
index e292e84db3c9..1b2e7e6cfdea 100644
--- a/crypto/rsa/Makefile
+++ b/crypto/rsa/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=rsa_test.c
diff --git a/crypto/seed/Makefile b/crypto/seed/Makefile
index 70d3d45a2be8..3440e2461f27 100644
--- a/crypto/seed/Makefile
+++ b/crypto/seed/Makefile
@@ -11,7 +11,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/sha/Makefile b/crypto/sha/Makefile
index de6cdde58a9c..a1c4175a1bf3 100644
--- a/crypto/sha/Makefile
+++ b/crypto/sha/Makefile
@@ -13,7 +13,7 @@ AR=             ar r
 
 SHA1_ASM_OBJ=
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/srp/Makefile b/crypto/srp/Makefile
index 414af7bc66b1..69a0bba7c2ae 100644
--- a/crypto/srp/Makefile
+++ b/crypto/srp/Makefile
@@ -12,7 +12,7 @@ MAKEDEPEND=	$(TOP)/util/domd $(TOP) -MD $(MAKEDEPPROG)
 MAKEFILE=	Makefile.ssl
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(INCLUDES) $(EXTRA_CFLAGS) $(CFLAG)
 
 GENERAL=Makefile
 TEST=srptest.c
diff --git a/crypto/stack/Makefile b/crypto/stack/Makefile
index b069c93237ae..420006fb5fa2 100644
--- a/crypto/stack/Makefile
+++ b/crypto/stack/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/store/Makefile b/crypto/store/Makefile
index 5bc7ca71f03c..a9a533831336 100644
--- a/crypto/store/Makefile
+++ b/crypto/store/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 #TEST= storetest.c
diff --git a/crypto/ts/Makefile b/crypto/ts/Makefile
index cf991efe4641..69f09c2c1543 100644
--- a/crypto/ts/Makefile
+++ b/crypto/ts/Makefile
@@ -17,8 +17,8 @@ AR=		ar r
 
 PEX_LIBS=
 EX_LIBS=
- 
-CFLAGS= $(INCLUDES) $(CFLAG)
+
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL= Makefile
 TEST=
diff --git a/crypto/txt_db/Makefile b/crypto/txt_db/Makefile
index 4f70b199a5d4..c500014a8c3f 100644
--- a/crypto/txt_db/Makefile
+++ b/crypto/txt_db/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/ui/Makefile b/crypto/ui/Makefile
index b28fcca6d965..2fb71ada399c 100644
--- a/crypto/ui/Makefile
+++ b/crypto/ui/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 #TEST= uitest.c
diff --git a/crypto/whrlpool/Makefile b/crypto/whrlpool/Makefile
index befd6d6f36ef..bca36fadc77e 100644
--- a/crypto/whrlpool/Makefile
+++ b/crypto/whrlpool/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 WP_ASM_OBJ=wp_block.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/x509/Makefile b/crypto/x509/Makefile
index bf197a1d932a..6cea5d58ff21 100644
--- a/crypto/x509/Makefile
+++ b/crypto/x509/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=verify_extra_test.c
diff --git a/crypto/x509v3/Makefile b/crypto/x509v3/Makefile
index 9791b77a0765..7e2207654119 100644
--- a/crypto/x509v3/Makefile
+++ b/crypto/x509v3/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=v3nametest.c
diff --git a/engines/Makefile b/engines/Makefile
index 2058ff405afe..cd2974b2b1a8 100644
--- a/engines/Makefile
+++ b/engines/Makefile
@@ -19,7 +19,7 @@ RECURSIVE_MAKE=	[ -z "$(ENGDIRS)" ] || for i in $(ENGDIRS) ; do \
 PEX_LIBS=
 EX_LIBS=
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile engines.com install.com engine_vector.mar
 TEST=
diff --git a/engines/ccgost/Makefile b/engines/ccgost/Makefile
index 17e1efbdff30..d7f1e93cd364 100644
--- a/engines/ccgost/Makefile
+++ b/engines/ccgost/Makefile
@@ -5,7 +5,7 @@ INCLUDES= -I../../include
 CFLAG=-g
 MAKEFILE= Makefile
 AR= ar r
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 LIB=$(TOP)/libcrypto.a
 
 LIBSRC= gost2001.c gost2001_keyx.c gost89.c gost94_keyx.c gost_ameth.c gost_asn1.c gost_crypt.c gost_ctl.c gost_eng.c gosthash.c gost_keywrap.c gost_md.c gost_params.c gost_pmeth.c gost_sign.c
diff --git a/ssl/Makefile b/ssl/Makefile
index b6dee5b5ea52..ff5dafc9043f 100644
--- a/ssl/Makefile
+++ b/ssl/Makefile
@@ -12,7 +12,7 @@ AR=		ar r
 # KRB5 stuff
 KRB5_INCLUDES=
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README ssl-lib.com install.com
 TEST=ssltest.c heartbeat_test.c clienthellotest.c sslv2conftest.c
diff --git a/test/Makefile b/test/Makefile
index e566babfa59d..a7955324534b 100644
--- a/test/Makefile
+++ b/test/Makefile
@@ -17,7 +17,7 @@ TEST=		igetest.c
 PEX_LIBS=
 EX_LIBS= #-lnsl -lsocket
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile maketests.com \
 	tests.com testenc.com tx509.com trsa.com tcrl.com tsid.com treq.com \
