diff --git a/src/meson.build b/src/meson.build
index 44c858c047d3..b5fbce110f0b 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -1007,7 +1007,7 @@ if have_gobject
   )
 
   gir = find_program('g-ir-scanner', required: get_option('introspection'))
-  build_gir = gir.found() and (not meson.is_cross_build() or get_option('introspection').enabled())
+  build_gir = gir.found() and get_option('introspection').allowed()
 
   build_gir = build_gir and get_option('default_library') != 'static'
   if not build_gir and get_option('introspection').enabled()
