diff --git a/Lib/distutils/sysconfig.py b/Lib/distutils/sysconfig.py
index b51629eb94f8..d8d14bc05cff 100644
--- a/Lib/distutils/sysconfig.py
+++ b/Lib/distutils/sysconfig.py
@@ -254,11 +254,15 @@ def get_makefile_filename():
     """Return full pathname of installed Makefile from the Python build."""
     if python_build:
         return os.path.join(_sys_home or project_base, "Makefile")
-    lib_dir = get_python_lib(plat_specific=0, standard_lib=1)
-    config_file = 'config-{}{}'.format(get_python_version(), build_flags)
-    if hasattr(sys.implementation, '_multiarch'):
-        config_file += '-%s' % sys.implementation._multiarch
-    return os.path.join(lib_dir, config_file, 'Makefile')
+    if 'PYTHON_HOST_CONFIG' in os.environ:
+        config_dir = os.environ['PYTHON_HOST_CONFIG']
+    else:
+        lib_dir = get_python_lib(plat_specific=0, standard_lib=1)
+        config_file = 'config-{}{}'.format(get_python_version(), build_flags)
+        if hasattr(sys.implementation, '_multiarch'):
+            config_file += '-%s' % sys.implementation._multiarch
+        config_dir = os.path.join(lib_dir, config_file)
+    return os.path.join(config_dir, 'Makefile')
 
 
 def parse_config_h(fp, g=None):
