diff --git a/uwsgiconfig.py b/uwsgiconfig.py
index 22c9dd34685d..67ee0ec2a02a 100644
--- a/uwsgiconfig.py
+++ b/uwsgiconfig.py
@@ -1074,24 +1074,36 @@ class uConf(object):
         # re-enable after pcre fix
         if self.get('pcre'):
             if self.get('pcre') == 'auto':
-                pcreconf = spcall('pcre-config --libs')
-                if pcreconf:
-                    self.libs.append(pcreconf)
-                    pcreconf = spcall("pcre-config --cflags")
-                    self.cflags.append(pcreconf)
+                libs = spcall('pkg-config --libs pcre')
+                if libs:
+                    cflags = spcall('pkg-config --cflags pcre')
+                else:
+                    libs = spcall('pcre-config --libs')
+                    if libs:
+                        cflags = spcall('pcre-config --cflags')
+
+                if libs and cflags:
+                    self.libs.append(libs)
+                    self.cflags.append(cflags)
                     self.gcc_list.append('core/regexp')
                     self.cflags.append("-DUWSGI_PCRE")
                     has_pcre = True
 
             else:
-                pcreconf = spcall('pcre-config --libs')
-                if pcreconf is None:
+                libs = spcall('pkg-config --libs pcre')
+                if libs:
+                    cflags = spcall('pkg-config --cflags pcre')
+                else:
+                    libs = spcall('pcre-config --libs')
+                    if libs:
+                        cflags = spcall('pcre-config --cflags')
+
+                if libs is None:
                     print("*** libpcre headers unavailable. uWSGI build is interrupted. You have to install pcre development package or disable pcre")
                     sys.exit(1)
                 else:
-                    self.libs.append(pcreconf)
-                    pcreconf = spcall("pcre-config --cflags")
-                    self.cflags.append(pcreconf)
+                    self.libs.append(libs)
+                    self.cflags.append(cflags)
                     self.gcc_list.append('core/regexp')
                     self.cflags.append("-DUWSGI_PCRE")
                     has_pcre = True
@@ -1305,11 +1317,17 @@ class uConf(object):
 
         if self.get('xml'):
             if self.get('xml') == 'auto':
-                xmlconf = spcall('xml2-config --libs')
-                if xmlconf and uwsgi_os != 'Darwin':
-                    self.libs.append(xmlconf)
-                    xmlconf = spcall("xml2-config --cflags")
-                    self.cflags.append(xmlconf)
+                libs = spcall('pkg-config --libs libxml-2.0')
+                if libs:
+                    cflags = spcall('pkg-config --cflags libxml-2.0')
+                else:
+                    libs = spcall('xml2-config --libs')
+                    if libs:
+                        cflags = spcall('xml2-config --cflags')
+
+                if libs and uwsgi_os != 'Darwin':
+                    self.libs.append(libs)
+                    self.cflags.append(cflags)
                     self.cflags.append("-DUWSGI_XML -DUWSGI_XML_LIBXML2")
                     self.gcc_list.append('core/xmlconf')
                     report['xml'] = 'libxml2'
@@ -1319,21 +1337,23 @@ class uConf(object):
                     self.gcc_list.append('core/xmlconf')
                     report['xml'] = 'expat'
             elif self.get('xml') == 'libxml2':
-                xmlconf = spcall('xml2-config --libs')
-                if xmlconf is None:
+                libs = spcall('pkg-config --libs libxml-2.0')
+                if libs:
+                    cflags = spcall('pkg-config --cflags libxml-2.0')
+                else:
+                    libs = spcall('xml2-config --libs')
+                    if libs:
+                        cflags = spcall('xml2-config --cflags')
+
+                if libs is None:
                     print("*** libxml2 headers unavailable. uWSGI build is interrupted. You have to install libxml2 development package or use libexpat or disable XML")
                     sys.exit(1)
                 else:
-                    self.libs.append(xmlconf)
-                    xmlconf = spcall("xml2-config --cflags")
-                    if xmlconf is None:
-                        print("*** libxml2 headers unavailable. uWSGI build is interrupted. You have to install libxml2 development package or use libexpat or disable XML")
-                        sys.exit(1)
-                    else:
-                        self.cflags.append(xmlconf)
-                        self.cflags.append("-DUWSGI_XML -DUWSGI_XML_LIBXML2")
-                        self.gcc_list.append('core/xmlconf')
-                        report['xml'] = 'libxml2'
+                    self.libs.append(libs)
+                    self.cflags.append(cflags)
+                    self.cflags.append("-DUWSGI_XML -DUWSGI_XML_LIBXML2")
+                    self.gcc_list.append('core/xmlconf')
+                    report['xml'] = 'libxml2'
             elif self.get('xml') == 'expat':
                 self.cflags.append("-DUWSGI_XML -DUWSGI_XML_EXPAT")
                 self.libs.append('-lexpat')
