diff --git a/configure.ac b/configure.ac
index fb7e52ebae33..097bcb1a92fc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -825,11 +825,7 @@ esac
 #
 # gpg-error is required.
 #
-AM_PATH_GPG_ERROR("$NEED_GPG_ERROR_VERSION")
-if test "x$GPG_ERROR_LIBS" = "x"; then
-  AC_MSG_ERROR([libgpg-error is needed.
-                See https://gnupg.org/ftp/gcrypt/gpgrt/ .])
-fi
+PKG_CHECK_MODULES([GPG_ERROR], [gpg-error >= $NEED_GPG_ERROR_VERSION])
 
 AC_DEFINE(GPG_ERR_SOURCE_DEFAULT, GPG_ERR_SOURCE_GCRYPT,
           [The default error source for libgcrypt.])
