diff --git a/setup.py b/setup.py
index cbc2641c59fc..03f470fc0440 100755
--- a/setup.py
+++ b/setup.py
@@ -13,6 +13,7 @@ import re
 import struct
 import subprocess
 import sys
+import sysconfig
 import warnings
 
 from setuptools import Extension, setup
@@ -234,6 +235,34 @@ def _cmd_exists(cmd):
         for path in os.environ["PATH"].split(os.pathsep)
     )
 
+def strip_and_clean(old_list):
+    new_list = []
+
+    for item in old_list:
+        item = item.strip()
+        if item:
+            new_list.append(item)
+
+    return new_list
+
+def sanitize_root(root):
+    if isinstance(root, list):
+        empty = True
+
+        for item in root:
+            if item:
+                empty = False
+                break
+
+        if empty:
+            root = []
+    else:
+        if root:
+            root = [ root ]
+        else:
+            root = []
+
+    return root
 
 def _pkg_config(name):
     try:
@@ -247,14 +276,18 @@ def _pkg_config(name):
             subprocess.check_output(command_libs)
             .decode("utf8")
             .strip()
-            .replace("-L", "")
+            .split("-L")
         )
         cflags = (
             subprocess.check_output(command_cflags)
             .decode("utf8")
             .strip()
-            .replace("-I", "")
+            .split("-I")
         )
+
+        cflags = strip_and_clean(cflags)
+        libs = strip_and_clean(libs)
+
         return (libs, cflags)
     except Exception:
         pass
@@ -396,8 +429,14 @@ class pil_build_ext(build_ext):
             else:
                 lib_root = include_root = root
 
-            _add_directory(library_dirs, lib_root)
-            _add_directory(include_dirs, include_root)
+            include_root = sanitize_root(include_root)
+            lib_root = sanitize_root(lib_root)
+
+            for root in lib_root:
+                _add_directory(library_dirs, root)
+
+            for root in include_root:
+                _add_directory(include_dirs, root)
 
         # respect CFLAGS/CPPFLAGS/LDFLAGS
         for k in ("CFLAGS", "CPPFLAGS", "LDFLAGS"):
@@ -418,8 +457,24 @@ class pil_build_ext(build_ext):
                 for d in os.environ[k].split(os.path.pathsep):
                     _add_directory(library_dirs, d)
 
-        _add_directory(library_dirs, os.path.join(sys.prefix, "lib"))
-        _add_directory(include_dirs, os.path.join(sys.prefix, "include"))
+        # check if we're cross-compiling
+        config = sysconfig.get_config_vars()
+        cross_compiling = False
+
+        if 'HOST_GNU_TYPE' in config and 'BUILD_GNU_TYPE' in config:
+            if config['HOST_GNU_TYPE'] != config['BUILD_GNU_TYPE']:
+                cross_compiling = True
+
+        if cross_compiling:
+            # add sysroot directories if one was specified
+            for match in re.finditer('--sysroot(\s+|=)([^\s]+)', config['PY_LDFLAGS']):
+                prefix = os.path.join(match.group(2), config['prefix'].lstrip(os.path.sep))
+
+                _add_directory(library_dirs, os.path.join(prefix, 'lib'))
+                _add_directory(include_dirs, os.path.join(prefix, 'include'))
+        else:
+            _add_directory(library_dirs, os.path.join(sys.prefix, "lib"))
+            _add_directory(include_dirs, os.path.join(sys.prefix, "include"))
 
         #
         # add platform directories
